app/Controllers/CategoryController.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace AgreableProductPlugin\Controllers;
+
+use Herbert\Framework\Models\Post;
+use Timber;
+use TimberPost;
+use Exception;
+use stdClass;
+
+class CategoryController {
+
+
+  /**
+   * Product category page
+   */
+  public function showProducts($product_collection_slug, $category_slug) {
+    if (!$product_collection = $this->get_products_by_category($product_collection_slug, $category_slug)) {
+      throw new Exception('Post not found');
+    }
+    
+    $context = Timber::get_context();
+    $context['product_collection'] = new TimberPost($product_collection);
+    $context['categories'] = $category_slug;
+
+    Timber::render('@AgreableProductPlugin/category.twig', $context, false);
+  }
+  
+  protected function get_products_by_category($product_collection_slug) {
```

Comment:
I don't think this function name reflects the code so its quite confusing.

"get_products_by_category", it seems to actually be getting a product collection based on the slug